sample/src/main/java/org/zalando/failsafeactuator/FailsafeSampleApp.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.boot.Banner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.core.env.Environment;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.zalando.failsafeactuator.service.FailsafeBreaker;
+
+import javax.annotation.PostConstruct;
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.util.concurrent.TimeUnit;
+
+@SpringBootApplication
+public class FailsafeSampleApp {
+
+    @RestController
+    public class HelloWorldController {
+        private boolean shouldFail = false;
+
+        @Autowired
+        @FailsafeBreaker(value = "testBreaker")
+        private CircuitBreaker breaker;
+
+        @Autowired
+        @FailsafeBreaker(value = "delayBreaker")
+        private CircuitBreaker delayBreaker;
+
+        @PostConstruct
+        private void init() {
+            delayBreaker.withDelay(5, TimeUnit.SECONDS);
+        }
+
+        @RequestMapping("/unreliable")
+        public String sayHelloWorldUnreliable() {
+            return getMessage();
+        }
+
+        @RequestMapping("/reliable")
+        public String sayHelloWorldReliable() {
+            return Failsafe.with(breaker).withFallback("Service unavailable").get(this::getMessage);
+        }
+
+        @RequestMapping("/reliableWithDelay")
+        public String sayHelloWorldReliableWithDelay() {
+            return Failsafe.with(delayBreaker).withFallback("Service unavailable").get(this::getMessage);
+        }
+
+        private String getMessage() {
+            if (shouldFail) {
+                shouldFail = false;
+                throw new RuntimeException("Every second try fails");
+            } else {
+                shouldFail = true;
+                return "Hello world!";
+            }
+        }
+
+    }
+
+    private static class FailsafeSampleAppBanner implements Banner {
+
+        @Override
+        public void printBanner(Environment environment, Class<?> sourceClass, PrintStream out) {
+            out.println("Failsafe-Actuator sample applicaton is running!");
+            out.println();
+            out.println("See the circuit breaker status:");
+            out.println("   $ curl http://127.0.0.1:8080/failsafe");
+            out.println("Unreliable endpoint that fails every second invocation:");
+            out.println("   $ curl http://127.0.0.1:8080/unreliable");
+            out.println("Endpoint with fallback:");
+            out.println("   $ curl http://127.0.0.1:8080/reliable");
+            out.println("Endpoint with fallback and 5 second circuit breaker delay:");
+            out.println("   $ curl http://127.0.0.1:8080/reliableWithDelay");
+        }
+
+    }
+
+    public static void main(final String[] args) {
```

Comment:
I would prefer having this in a separate class. Would you mind moving this?

Reply:
I've broken up the sample application into a main, controller and banner part. What do you think of it now?